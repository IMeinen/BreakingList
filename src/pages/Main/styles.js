import styled from 'styled-components';

export const LogoIMG = styled.img`
  display: block;
  margin-top: 30px;
  margin-left: auto;
  margin-right: auto;
  width: 200px;
  height: 125px;

  @media only screen and (max-width: 468px) {
    width: 150px;
    height: 93px;
  }
`;

export const HintText = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  color: #fff;
  margin: 15px 0px 15px 0px;

  @media only screen and (max-width: 468px) {
    h2 {
      font-size: 0.7em;
    }
  }
`;

export const ListContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-start;
  flex-wrap: wrap;
  margin-top: 30px;
  margin-left: auto;
  margin-right: auto;
  width: 1320px;
  background-color: transparent;
  @media only screen and (min-width: 1201px) and (max-width: 1500px) {
    width: 80%;
  }
  @media only screen and (max-width: 1200px) {
    width: 90%;
    margin-top: 15px;
    justify-content: center;
  }


`;

