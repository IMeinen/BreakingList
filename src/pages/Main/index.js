import React, { useState, useEffect, useCallback } from 'react';

import Logo from '../../images/logo.png';
import { LogoIMG, ListContainer, HintText } from './styles';
import FilterInput from '../../components/FilterInput';
import Modal from '../../components/Modal';
import Overlay from '../../components/Overlay';
import ListComponent from './Components/ListComponent';
import Loading from '../../components/Loading';

import { api } from '../../services/api';

export default function Main() {
  const [characters, setCharacters] = useState();
  const [currentCharacters, setCurrentCharacters] = useState();
  const [openModal, setOpenModal] = useState(false);
  const [selectedChar, setSelectedChar] = useState();
  const [loading, setLoading] = useState(true);

  const handleLoad = useCallback(async () => {
    const [error, response] = await api.getCharactersData();

    if (response) {
      setCurrentCharacters(response);
      setCharacters(response);
      setLoading(false);
    }
    if (error) {
      console.log(error.message);
    }
  },[]);

  useEffect(() => {
    handleLoad();
  }, [handleLoad]);

  return (
    <div style={{ maxWidth: '100%', width: '100%' }}>
      <LogoIMG src={Logo} alt="logo_image" />
      <HintText>
        <h2>Clique nos personagens para visualizar suas informações.</h2>
      </HintText>
      {openModal && (
        <Overlay>
          <Modal charInfo={selectedChar[0]} modalOpened={setOpenModal} />
        </Overlay>
      )}
      {!loading && (
        <ListContainer>
          {currentCharacters && (
            <FilterInput data={characters} set={setCurrentCharacters} />
          )}

          <ListComponent
            characters={characters}
            currentCharacters={currentCharacters}
            setSelectedChar={setSelectedChar}
            setOpenModal={setOpenModal}
          />
        </ListContainer>

      )}
      {loading && (<Loading /> )}
    </div>
  );
}
