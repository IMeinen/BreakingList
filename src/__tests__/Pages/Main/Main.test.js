import React from 'react';

import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Main from '../../../pages/Main'

Enzyme.configure({ adapter: new Adapter() });
describe('Main page tests', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Main/>);
    expect(wrapper).toMatchSnapshot();
  });
});
