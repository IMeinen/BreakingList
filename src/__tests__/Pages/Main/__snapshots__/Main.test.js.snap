// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page tests renders correctly 1`] = `
<div
  style={
    Object {
      "maxWidth": "100%",
      "width": "100%",
    }
  }
>
  <styled.img
    alt="logo_image"
    src="logo.png"
  />
  <styled.div>
    <h2>
      Clique nos personagens para visualizar suas informações.
    </h2>
  </styled.div>
  <Loading />
</div>
`;
